api/src/main/java/org/apache/iceberg/PartitionSpec.java

```diff
@@ -140,6 +142,30 @@ public StructType partitionType() {
     return lazyPartitionType;
   }
 
+  /**
+   * Returns a struct with TransformID's which match the ID's used in the Table Metadata. This is
```

Comment:
Second sentence is a bit confusing here. Maybe instead

"See {link #partitionType} for a struct with reassigned transformIds that will not overlap with any defined column fieldIds."